app/test_helpers.go

```diff
@@ -0,0 +1,161 @@
+package app
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/b9lab/checkers/x/checkers/types"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+)
+
+var DefaultConsensusParams = &abci.ConsensusParams{
+	Block: &abci.BlockParams{
+		MaxBytes: 200000,
+		MaxGas:   2000000,
+	},
+	Evidence: &tmproto.EvidenceParams{
+		MaxAgeNumBlocks: 302400,
+		MaxAgeDuration:  504 * time.Hour, // 3 weeks is the max duration
+		MaxBytes:        10000,
+	},
+	Validator: &tmproto.ValidatorParams{
+		PubKeyTypes: []string{
+			tmtypes.ABCIPubKeyTypeEd25519,
+		},
+	},
+}
+
+func setup(withGenesis bool, invCheckPeriod uint) (*App, GenesisState) {
+	db := dbm.NewMemDB()
+	encCdc := MakeTestEncodingConfig()
```

Comment:
The tutorial bases on [ignite 0.17.3](https://github.com/ignite/cli/blob/v0.17.3/go.mod#L47), which itself uses [tendermint/spm](https://github.com/tendermint/spm/commit/42e72ad05b63df369b679d64e8cfedaea15ba1c9).
Adding `import "github.com/tendermint/spm/cosmoscmd"` and replacing this line by `encCdc := cosmoscmd.MakeEncodingConfig(ModuleBasics)` should behave the same, if I understood the code correctly.

If my assumption is correct, this change would simplify the tutorial, because app/encoding.go and app/param/proto.go could be omitted.

With the current [ignite version](https://github.com/ignite/cli/blob/v0.23.0/ignite/pkg/cosmoscmd/encoding.go),
the import has to be changed to `"github.com/ignite/cli/ignite/pkg/cosmoscmd"`

Reply:
Takes note. Thank you.